transport/nio/src/main/java/org/kaazing/gateway/transport/nio/internal/AbstractNioConnector.java

```diff
@@ -143,79 +147,47 @@ public void connectDestroy(ResourceAddress address) {
-        ConnectFuture future;
-
         // TODO: throw exception if address contains more then one resource
         ResourceAddress transport = address.getTransport();
         if (transport != null) {
-            final DefaultConnectFuture bridgeConnectFuture = new DefaultConnectFuture();
-
-            // propagate connection failure, if necessary
-            IoFutureListener<ConnectFuture> parentConnectListener = new IoFutureListener<ConnectFuture>() {
-                @Override
-                public void operationComplete(ConnectFuture future) {
-                    // fail bridge connect future if parent connect fails
-                    if (!future.isConnected()) {
-                        bridgeConnectFuture.setException(future.getException());
-                    }
-                }
-            };
-
-            final BridgeConnector connector = bridgeServiceFactory.newBridgeConnector(transport);
-
-            IoSessionInitializer<ConnectFuture> parentInitializer = createParentInitializer(address,
-                                                                                            handler,
-                                                                                            (IoSessionInitializer)initializer,
-                                                                                            bridgeConnectFuture);
-
-            connector.connect(transport, tcpBridgeHandler, parentInitializer).addListener(parentConnectListener);
-
-            return bridgeConnectFuture;
+            return getTransportConnectFuture(address, handler, initializer, transport);
         }
 
         final URI resource = address.getResource();
         final InetSocketAddress inetAddress = new InetSocketAddress(resource.getHost(), resource.getPort());
-
         if (logger.isTraceEnabled()) {
         	logger.trace(format("AbstractNioConnector.connectInternal(), resource: %s", resource));
         }
 
         // KG-1452: Avoid deadlock between dispose and connectInternal
-        IoConnector connector = this.connector.get();
+        IoConnector connector = this.connectorReference.get();
         if (connector == null) {
             return DefaultConnectFuture.newFailedFuture(new IllegalStateException("Connector is being shut down"));
         }
 
         final String nextProtocol = address.getOption(ResourceAddress.NEXT_PROTOCOL);
-
-        future = connector.connect(inetAddress, new IoSessionInitializer<T>() {
+        ConnectFuture future = connector.connect(inetAddress, new IoSessionInitializer<T>() {
             @Override
             public void initializeSession(IoSession session, T future) {
-
                 if (logger.isTraceEnabled()) {
                     logger.trace(format("AbstractNioConnector.connectInternal()$initializeSession(), session: %s, resource: %s", session, resource));
                 }
-
+                registerConnectFilters(address, session);
```

Comment:
I'm ok with adding the register*Filters, But I'm not overly familiar with this code.  So I would want to defer to @jitsni judgment.  

Another alternative would be to make the "align" feature not UDP specific.  Then only expose public configuration to set it on udp.  That would eliminate the need for this new API.  I probably would have done it that way initially but there are tradeoffs with both the approaches mentioned here.